lib/reportportal.rb

```diff
@@ -48,17 +48,20 @@ def now
     end
 
     def status_to_level(status)
-      LOG_LEVELS.fetch(status)
-      # case status
-      # when :failed, :undefined, :pending
-      #   LOG_LEVELS[:error]
-      # when :skipped
-      #   LOG_LEVELS[:warn]
-      # when :passed
-      #   LOG_LEVELS[:info]
-      # else
-      #   LOG_LEVELS.values.include?(status) ? status : LOG_LEVELS[:unknown]
-      # end
+      case status
+        when :passed
+          LOG_LEVELS[:info]
+        when :failed, :undefined, :pending, :error
+          LOG_LEVELS[:error]
+        when :skipped
+          LOG_LEVELS[:warn]
+        when :unknown
```

Comment:
`:trace` is also provided [in some place](https://github.com/reportportal/agent-ruby/blob/447d93bf06a874df65204926a4c7bba7f7f4d19d/lib/report_portal/cucumber/report.rb#L103).

I think this code can be simplified to:
```ruby
case status
when :passed
  LOG_LEVELS[:info]
when :failed, :undefined, :pending
  LOG_LEVELS[:error]
when :skipped
  LOG_LEVELS[:warn]
else
  LOG_LEVELS.fetch(status, :unknown)
end
```

Also please use indentation as described in Ruby style guide - ["Indent when as deep as case"](https://github.com/bbatsov/ruby-style-guide#indent-when-to-case).

Reply:
I think such "else" part would be unsafe.
If status value somehow will be different from the keys of LOG_LEVELS hash whole test harness will fail with uncaught error.

Reply:
Ok, I understand.
But it's also not good to report any custom log levels as `:info`.
Users may add custom log levels as `ReportPortal::LOG_LEVELS[:custom] = 'INFO'`.
`LOG_LEVELS.fetch(status)` supports such use case.
A case statement like in this PR does not really support custom log levels.